compiler/plugins/target/AMD-AIE/iree-amd-aie/Transforms/AMDAIESplitControlPacketData.cpp

```diff
@@ -87,7 +87,10 @@ void AMDAIESplitControlPacketDataPass::runOnOperation() {
   }
   AMDAIE::AMDAIEDeviceModel deviceModel =
       AMDAIE::getDeviceModel(maybeDevice.value());
-  uint32_t maxLength = deviceModel.getCtrlPktMaxLength();
+  // Add 1 to the `beat` field value, as it represents `length - 1` in the
+  // control packet header. For example, if the maximum `beat` field value is
+  // 15, the corresponding maximum length is 16.
+  uint32_t maxLength = deviceModel.getCtrlPktMaxBeat() + 1;
```

Comment:
I think `getCtrlPktMaxLength` is more descriptive, so could we keep it like that and do the `+ 1` addition inside that function?